engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -1461,6 +1460,7 @@ else if ( propertyPathNode.getKey() != null ) {
 
 		metaConstraints.addAll( propertyMetaData.getConstraints() );
 		typeArgumentConstraints.addAll( propertyMetaData.getTypeArgumentsConstraints() );
+		propertyPath.removeLeafNode();
```

Comment:
OK. Could you add a comment at the end of the `propertyPath.removeLeafNode();` lines so that we remember why we remove this last node?

It wasn't obvious when I read the code so I'm pretty sure it won't be in 2 years when I'll read that code again!